core/test/common/MultiRole.js

```diff
@@ -137,6 +142,11 @@ contract("MultiRole", function(accounts) {
     await multiRole.removeMember("2", account2, { from: account1 });
     assert.isFalse(await multiRole.holdsRole("2", account2));
     assert(await didContractThrow(multiRole.revertIfNotHoldingRole("2", { from: account2 })));
+
+    // Anyone who who holds the managing role can renounce their membership.
```

Comment:
I think `L03` is referring to the case where the role holder *does not* hold the managing role. Renouncing a role is basically a special case where the holder of a role should be able to give up their own membership, even if they don't have the power to modify anyone else's membership (or add themselves back). Does that make sense? Or do you think I'm misinterpreting the OZ issue?

Reply:
I think this special case is correct. An `ExclusiveRole` must have a separate RoleManager from a RoleHolder. In this situation, they are recommending we add an ability for the RoleHolder to renounce their membership.

Reply:
For `SharedRoles` the caller who wants to renounce their membership can just call `removeMember` and pass in their own address. So I think we should just add a function renounces membership specifically for ExclusiveRoles

Reply:
I made the changes we've discussed in this thread now

Reply:
Wait, are you sure about the SharedRole?

Let's say `address1` is a member of `role1` (exclusive) and `address2` is a member of `role2` (shared). `role1` manages `role2`. My understanding is that `address2` cannot remove itself from `role2`. `address1` is the only one with permission to do that. Is that not the case?

Reply:
@mrice32 Ah yes, I didn't word that correctly. So we actually should add a way for non-managers to renounce their shared role membership

Reply:
Ok I fixed this. Now both exclusive and shared roles can be renounced by any holder via `renounceMembership`